pkg/controllers/coscheduling/podgroup_controller.go

```diff
@@ -18,17 +18,36 @@ package controllers
 
 import (
 	"context"
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+	"time"
 
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 
-	v1alpha1 "sigs.k8s.io/scheduler-plugins/apis/scheduling/v1alpha1"
+	schedv1alpha1 "sigs.k8s.io/scheduler-plugins/apis/scheduling/v1alpha1"
+	"sigs.k8s.io/scheduler-plugins/pkg/util"
 )
 
 // PodGroupReconciler reconciles a PodGroup object
 type PodGroupReconciler struct {
+	log      logr.Logger
+	recorder record.EventRecorder
```

Comment:
this is unused you can remove it.

Reply:
`recorder` will be used in later development, and it is worthwhile to add it here when we first migrate to controller runtime in this controller runtime way

Reply:
You can add TODO here.